e2e/pod.go

```diff
@@ -623,3 +624,82 @@ func verifySeLinuxMountOption(
 
 	return nil
 }
+
+// verifyReadAffinity verifies if read affinity is enabled by checking if read_from_replica
+// and crush_location options are present in the device config file (/sys/devices/rbd/0/config_info).
+func verifyReadAffinity(
+	f *framework.Framework,
+	pvcPath, appPath, daemonSetName, cn, ns string,
+) error {
+	readFromReplicaOption := "read_from_replica=localize"
+	expectedCrushLocationValues := map[string]string{
+		strings.Split(crushLocationRegionLabel, "/")[1]: crushLocationRegionValue,
+		strings.Split(crushLocationZoneLabel, "/")[1]:   crushLocationZoneValue,
+	}
+
+	// create PVC
+	pvc, err := loadPVC(pvcPath)
+	if err != nil {
+		return fmt.Errorf("failed to load pvc: %w", err)
+	}
+	pvc.Namespace = f.UniqueName
+	err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to create PVC: %w", err)
+	}
+	app, err := loadApp(appPath)
+	if err != nil {
+		return fmt.Errorf("failed to load application: %w", err)
+	}
+	app.Namespace = f.UniqueName
+	err = createApp(f.ClientSet, app, deployTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to create application: %w", err)
+	}
+
+	selector, err := getDaemonSetLabelSelector(f, ns, daemonSetName)
+	if err != nil {
+		return fmt.Errorf("failed to get selector label")
+	}
+
+	opt := metav1.ListOptions{
+		LabelSelector: selector,
+	}
+	command := "cat /sys/devices/rbd/0/config_info"
```

Comment:
we cant have `0` as hardcoded value, we need to get the device number from mapped image.

Reply:
Okay, I thought during each e2e test case we will be having only one device so will use `0`.

Reply:
@Madhu-1
```
bash-4.4$ rbd device list --format=json
[{"id":"0","pool":"replicapool","namespace":"","name":"csi-vol-37f39fc8-e26d-40f3-a297-acd16e7afbb8","snap":"-","device":"/dev/rbd0"}]
``` 

does this `id` refer to device id?